### PR TITLE
RADIUS module update.

### PR DESCRIPTION
**Pull request by plorinquer**
Hello,

This PR updates the ``radius`` module, notably by adding the ability to create specific RADIUS attributes, and to compute the ``authenticator`` field in RADIUS packets, as well as the ``Message-Authenticator`` attribute value.

Since I have made a number of changes, I allowed myself to edit previous lines of code in order to make the module "more" PEP 8 compliant.

### CODE REVIEW

File: scapy/layers/radius.py [hunk 1]
```diff
@@ -9,139 +9,1190 @@
...
+            attr_type = ord(_pkt[0])
+            return cls.registered_attributes.get(attr_type, cls)
+        return cls
+
+    def haslayer(self, cls):
+        ret = 0
+        if cls == RadiusAttribute:
+            for attr_class in RadiusAttribute.registered_attributes.values():
+                if isinstance(self, attr_class):
+                    ret = 1
```
Comment (guedou) [inline scapy/layers/radius.py]: use return True instead

File: scapy/layers/radius.py [hunk 2]
```diff
@@ -9,139 +9,1190 @@
...
+                    break
+        elif cls in RadiusAttribute.registered_attributes.values() and isinstance(self, cls):
+            ret = 1
```
Comment (guedou) [inline scapy/layers/radius.py]: use return True

File: scapy/layers/radius.py [hunk 3]
```diff
@@ -9,139 +9,1190 @@
...
+        return ret
```
Comment (guedou) [inline scapy/layers/radius.py]: use return False

File: scapy/layers/radius.py [hunk 4]
```diff
@@ -9,139 +9,1190 @@
...
+    ):
+        """
+        Computes the "Message-Authenticator" of a given RADIUS packet.
+        """
+
+        if not conf.crypto_valid:
+            g_log_loading.info(_crypto_loading_failure_message)
+            return None
+
+        packed_hdr = struct.pack("!B", radius_packet.code) +\
```
Comment (guedou) [inline scapy/layers/radius.py]: using packet_hdr += will make the code easier to read

File: scapy/layers/radius.py [hunk 5]
```diff
@@ -9,139 +9,1190 @@
...
+    def compute_authenticator(self, packed_request_auth, shared_secret):
+        """
+        Computes the authenticator field (RFC 2865 - Section 3)
+        """
+
+        if not conf.crypto_valid:
+            g_log_loading.info(_crypto_loading_failure_message)
+            return None
+
+        packed_hdr = struct.pack("!B", self.code) +\
```
Comment (guedou) [inline scapy/layers/radius.py]: same here

**Comment by gpotter2** [comment]
Doing the background research for the advisory.
So this PR got us a CVE :smile:

**Comment by guedou** [comment]
I believe that other similar issues were introduced and fixed during Scapy life =)

**Pull request merged**